test/setup/MFCompilerTest.php

```diff
@@ -123,15 +136,57 @@ public function UseLatestPrecompiledFileProvider(){
 		];
 	}
 
-	private function BuildProviderUseCaseArray(string $sPrecompiledFileUri, string $sPostCompilationLatestPrecompiledFile, $sExpectedReturn) : array{
+	private function BuildProviderUseCaseArray(string $sPrecompiledFileUri, string $sPostCompilationLatestPrecompiledFile, $sExpectedReturn, $bDisableThemePrecompilationViaConf = false) : array{
 		return [
 			"sTempTargetDir" => sys_get_temp_dir(),
 			"sPrecompiledFileUri" => $sPrecompiledFileUri,
 			"sPostCompilationLatestPrecompiledFile" => $sPostCompilationLatestPrecompiledFile,
 			"sThemeDir" => "test",
-			"sExpectedReturn" => $sExpectedReturn
+			"sExpectedReturn" => $sExpectedReturn,
+			"bDisableThemePrecompilationViaConf" => $bDisableThemePrecompilationViaConf
 		];
 	}
 
+	public function testCompileThemes(){
+		$sXmlDataCustoFilePath = realpath(__DIR__ . '/ressources/datamodels/datamodel-branding.xml');
+		$oDom = new MFDocument();
+		$oDom->load($sXmlDataCustoFilePath);
+
+		/** @var \MFElement $oBrandingNode */
+		$oBrandingNode = $oDom->GetNodes('branding')->item(0);
+
+		$this->RecurseMkdir($this->sTmpDir.'/branding/themes/fullmoon/');
+		file_put_contents($this->sTmpDir.'/branding/themes/fullmoon/main.css', "");
+
+		$aImportsPaths = array(
+			APPROOT.'css/',
+			APPROOT.'css/backoffice/main.scss',
+			$this->sTmpDir.'//',
+		);
+
+
+		$aThemeParameters = [
+			'variables' => [
+				'ibo-page-banner--background-color' => '$ibo-color-red-600',
+				'ibo-page-banner--text-color' => '$ibo-color-red-100',
+				'ibo-page-banner--text-content' => '"THIS IS A TEST INSTANCE"',
+			],
+			'imports_variable' => [ 'style2' => 'style2.scss'],
+			'imports_utility' => [ 'style1' => 'style1.scss', 'style3' => 'style3.scss'],
+			'stylesheets' => [
+				"fullmoon" => '../css/backoffice/main.scss',
+                "environment-banner" => '../css/backoffice/themes/page-banner.scss'
+			],
+		];
+
+		$oThemeHandlerService = $this->createMock(\ThemeHandlerService::class);
+		$oThemeHandlerService->expects($this->exactly(1))
+			->method("CompileTheme")
+			->with("fullmoon", true, $this->oMFCompiler->GetCompilationTimeStamp(), $aThemeParameters, $aImportsPaths, $this->sTmpDir . '/');
 
+
+		//CompileTheme($sThemeId, $bSetup = false, $sSetupCompilationTimestamp="", $aThemeParameters = null, $aImportsPaths = null, $sWorkingPath = null)
+		MFCompiler::setThemeHandlerService($oThemeHandlerService);
```

Comment:
```suggestion
		MFCompiler::SetThemeHandlerService($oThemeHandlerService);
```

Rename to be up to date with the previous change